vars/buildDockerAndPublishImage.groovy

```diff
@@ -217,15 +262,24 @@ def call(String imageShortName, Map userConfig=[:]) {
                 imageDeployName += ":${env.TAG_NAME}"
               }
             }
-
-            withEnv(["IMAGE_DEPLOY_NAME=${imageDeployName}"]) {
-              // Please note that "make deploy" uses the environment variable "IMAGE_DEPLOY_NAME"
-              if (isUnix()) {
-                sh 'make deploy'
-              } else {
-                powershell 'make deploy'
-              }
-            } // withEnv
+            if (isUnix()) {
+              withEnv(["IMAGE_DEPLOY_NAME=${imageDeployName}"]) {
+                // Please note that "make deploy" and the generated bake deploy file uses the environment variable "IMAGE_DEPLOY_NAME"
+                if (finalConfig.dockerBakeFile != '') {
+                  sh 'make bake-deploy'
+                } else {
+                  if (operatingSystem == "linux") {
+                    //linux ==> generated docker bake
+                    sh 'make bake-deploy'
+                  } else {
+                    // old process still used for windows
+                    sh 'make deploy'
+                  }
```

Comment:
Improvement proposal : factorize this set of conditions as it is a duplicated code leading to error (e.g. forgetting to update `deploy` when update `build` logic)